tests/conf/nasa_marine_debris.yaml

```diff
@@ -1,7 +1,7 @@
 experiment:
   task: "nasa_marine_debris"
   module:
-    detection_model: "faster-rcnn"
+    model: "faster-rcnn"
     backbone: "resnet18"
```

Comment:
I think we decided to call this "encoder"?

Reply:
Oh I understood, that like in segmentation where the Unet is the model, for Object detection the faster-rcnn would be the model. So you want to go detection_model -> encoder? It kind of feels together with "backbone".

Reply:
No, what I meant was "detection_model" -> "model" is correct, but shouldn't it be "backbone" -> "encoder"?

Reply:
https://github.com/microsoft/torchgeo/pull/854#issuecomment-1328656114 maybe to summarize it all together.

Reply:
Doesn't "encoder" imply there is a "decoder"? Segmentation models follow an encoder decoder architecture where the pre-trained backbone (ResNet, VGG etc) acts as the encoder. I believe that's why `segmentation.models` use that term. But in other tasks I've seen them usually referred to as backbone.

Reply:
Maybe we should use "model" and "backbone" everywhere instead of "model" and "encoder"? Are there any models that let you replace both the "encoder" and "decoder" separately?

Reply:
> Are there any models that let you replace both the "encoder" and "decoder" separately?

Not that I know of -- the decoder part of models in `smp` is fixed.

"backbone" is a good choice

Reply:
I think there's two questions here:

1. Can we use the same name for all sub-models?
2. Should we call them encoders or backbones?

I agree that encoder kind of implies decoder, which won't be the case for all models. Would backbone be a better more generic name for what we're describing?

Reply:
no, they are both good choices -- I lean towards encoder because it is more specific to me, but I don't have a strong opinion as long as we are consistent. An encoder to me is any architecture that produces a latent representation